runtime/src/main/java/org/corfudb/util/MetricsUtils.java

```diff
@@ -251,4 +240,71 @@ public static void incConditionalCounter(boolean enabled, @NonNull Counter count
             counter.inc(amount);
         }
     }
+
+    /**
+     * This method creates object size gauge and registers it to the metrics registry. The
+     * method guarantees that no strong reference to the provided object will be retained.
+     * Hence it does not prevent garbage collection if the client does not hold a strong
+     * reference to the object being measured.
+     *
+     * @param metrics the metrics registry to which the size gauge will be registered.
+     * @param object the object which its size will be measured with a corresponding
+     *               gauge*
+     *
+     * @return a gauge that provides an estimate of deep size for the provided object
+     *         as long as there exist a strong reference to that object.
+     */
+    public static Gauge<Long> addMemoryMeasurerFor(@NonNull MetricRegistry metrics,
+                                                   Object object) {
+        String simpleClassName = object == null ?
+                "Object" :
+                object.getClass().getSimpleName();
+
+        return  metrics.register(String.format("deep-sizeof.%s@%04x",
+                                               simpleClassName,
+                                               System.identityHashCode(object)),
+                                 getSizeGauge(object));
+    }
+
+    /**
+     * This method creates a gauge that returned the deep size estimation of an object
+     * while there is a reference to it in the code. Otherwise it will return 0 as the
+     * object can be claimed by Garbage collection. Note that this gauge guarantees
+     * that it will not hold any strong reference to the object that it measures.
+     *
+     * @param object an object to which its size will be estimated using the return
+     *               gauge
+     * @return a gauge that provides an estimate of deep size for the provided object
+     *         as long as there exist a strong reference to that object.
+     */
+    private static Gauge<Long> getSizeGauge(Object object) {
+        WeakReference<Object> toBeMeasuredObject = new WeakReference<>(object);
+
+        return new Gauge<Long>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)